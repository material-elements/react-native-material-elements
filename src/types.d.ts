export * from './components/types';
export * from './types/index';
