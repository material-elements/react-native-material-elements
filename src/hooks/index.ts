export * from './useRestyle';
export * from './useThemedProps';
