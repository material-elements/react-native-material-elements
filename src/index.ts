export * from './components';
export * from './hooks';
export * from './libraries';
export * from './utils';
