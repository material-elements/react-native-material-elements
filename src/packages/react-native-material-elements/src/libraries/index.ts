export * from './themes';
export * from './style';
