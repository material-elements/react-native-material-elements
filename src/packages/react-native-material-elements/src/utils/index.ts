export * from './styleGenerator';
export * from './utils';
export * from './dimensions';
