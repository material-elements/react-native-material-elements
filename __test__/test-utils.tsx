import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

export const ThemeWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: ThemeWrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
