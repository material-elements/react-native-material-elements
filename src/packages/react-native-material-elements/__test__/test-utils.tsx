import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';
import { ThemeProviderProps } from '../src/libraries/types';

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  themeProps?: Partial<ThemeProviderProps>;
}

export const ThemeWrapper: React.FC<ThemeProviderProps> = ({ children, ...props }) => {
  return <ThemeProvider {...props}>{children}</ThemeProvider>;
};

export const customRender = (ui: React.ReactElement, options?: CustomRenderOptions) => {
  const { themeProps, ...rest } = options || {};
  return render(ui, {
    wrapper: wrapperProps => <ThemeWrapper {...themeProps}>{wrapperProps.children}</ThemeWrapper>,
    ...rest,
  });
};

export * from '@testing-library/react-native';
export { customRender as render };
