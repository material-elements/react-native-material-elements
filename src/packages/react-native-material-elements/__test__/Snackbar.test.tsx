import { DeviceEventEmitter } from 'react-native';
import { HIDE_SNACK_BAR_MESSAGE, SHOW_SNACK_BAR_MESSAGE, Snackbar, snackbar, SnackbarProperties } from '../src';
import { render } from './test-utils';

describe('snackbar utils', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render with default props', () => {
    const { toJSON } = render(<Snackbar />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should called the snack bar show method with config', () => {
    snackbar.show({} as any);
    expect(DeviceEventEmitter.emit).toHaveBeenCalled();
  });

  it('should called the snack bar show method with snack properties config', () => {
    const config: SnackbarProperties = {
      message: 'Marked as read',
      showActionButton: true,
      shouldHideWhenClickedOnActionButton: false,
      type: 'warning',
    };
    snackbar.show(config);
    expect(DeviceEventEmitter.emit).toHaveBeenCalledTimes(1);
    expect(DeviceEventEmitter.emit).toHaveBeenCalledWith(SHOW_SNACK_BAR_MESSAGE, config);
  });

  it('should called the snack bar hide method', () => {
    snackbar.hide();
    expect(DeviceEventEmitter.emit).toHaveReturnedTimes(1);
    expect(DeviceEventEmitter.emit).toHaveBeenCalledWith(HIDE_SNACK_BAR_MESSAGE);
  });
});
