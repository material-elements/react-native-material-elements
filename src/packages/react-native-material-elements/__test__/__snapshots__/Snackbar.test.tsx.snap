// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snackbar utils should render with default props 1`] = `null`;
